### PR TITLE
feat(transfer): add manual channel with external node

### LINKED ISSUE #1563: [Bug]: Prevent user from buying channel when node capacity is reached
### Describe the bug

If the node has reached it's (legal) upper limit for capacity the LSP will throw an error like:

```
Channel open failed. {"message": "Node capacity is above our capacity limit. Contact us for more information.", "type": "CHANNEL_REJECTED_BY_LSP", "detail": {}, "name": "ChannelOpenError", "grenacheCall": {"workerName": "blocktank-lsp-ln2", "method": "orderChannel", "args":["<nodeId>", true, 1141097, {"pushBalanceSat": 1234567, "lspNodeId": "0296b2db342fcf87ea94d981757fdf4d3e545bd5cef4919f58b5d38dfdd73bf5c9", "zeroConf": false, "zeroReserve": true}], "options": {"timeoutMs":60000}}}
```

Bitkit should do some checks and prevent the user from buying addtional (large) channels.

### Screenshots / Recording

_No response_

### Operating system

Android, iOS

### Bitkit version

master

### Log output

_No response_

### LINKED ISSUE #1972: [Feature]: LN manual channel with external node
### Describe the problem

Blocked by https://github.com/synonymdev/react-native-ldk/issues/30

While Bitkit does support LNURL-channel, there is no way to connect to nodes that do not generate such a QR.

### Describe the solution

Add the flow for adding non Blocktank nodes. Also need to make sure open/close transactions are marked as such in the activity.

image

### PR DESCRIPTION
**Pull request by pwltr**
### Description

Adds locally funded channel opening with external node.

### Known Issues
- Transfer to spending pending time doesn't update until channel is opened
- Channel closes always end in force close because of fee consensus, test mainnet (`ProcessingError`)

### Linked Issues/Tasks

Closes https://github.com/synonymdev/bitkit/issues/1972

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (improving code without creating new functionality)

### Tests

- [x] Detox test
- [ ] Unit test
- [ ] No test

### Screenshot / Video

https://github.com/user-attachments/assets/f5e69975-5ba1-42c6-b904-44197c0ec096

### QA Notes

With an onchain balance, open the scanner and paste a node connection URI. For regtest can use the Blocktank node: `03b9a456fb45d5ac98c02040d39aec77fa3eeb41fd22cf40b862b393bcfc43473a@35.233.47.252:9400`

### CODE REVIEW

File: src/screens/Transfer/ExternalNode/Amount.tsx [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+		return convertToSats(textFieldValue, conversionUnit);
+	}, [textFieldValue, conversionUnit]);
+
+	const availableAmount = useMemo(() => {
+		const maxAmountResponse = getMaxSendAmount();
+		if (maxAmountResponse.isOk()) {
+			return maxAmountResponse.value.amount;
+		}
+		return 0;
+		// eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (limpbrains) [inline src/screens/Transfer/ExternalNode/Amount.tsx:87]: please, leave a comment of what is excluded from deps and why

File: src/utils/i18n/locales/de/wallet.json [hunk 2]
```diff
@@ -363,7 +363,7 @@
     "string": "Zum Ausgabenkonto"
   },
   "activity_transfer_pending": {
-    "string": "Übertragung (±{duration}m)"
```
Comment (limpbrains) [inline src/utils/i18n/locales/de/wallet.json:366]: this will be overwritten, if you will not update translation on Transifex

Comment (pwltr) [reply src/utils/i18n/locales/de/wallet.json:366]: that's fine

File: src/screens/Transfer/ExternalNode/Amount.tsx (hunk 1 above)
Comment (pwltr) [reply src/screens/Transfer/ExternalNode/Amount.tsx:87]: added a comment

**Review by coreyphillips — APPROVED** [review]

File: e2e/channels.e2e.js [hunk 3]
```diff
@@ -46,134 +48,295 @@ d('LN Channel Onboarding', () => {
...
+		await element(by.id('NavigationClose')).tap();
+		await element(by.id('WalletsScrollView')).scrollTo('top', NaN, 0.85);
+
+		// Mine 3 blocks
+		await rpc.generateToAddress(3, await rpc.getNewAddress());
+
+		// TODO: mine single blocks and check updated transfer time
+
+		// wait for channel to be opened
+		await sleep(5000);
```
Comment (limpbrains) [inline e2e/channels.e2e.js:277]: Maybe use lnd api here? should be more stable.

**Review by limpbrains — APPROVED** [review]

**Pull request merged**